### ISSUE TITLE
ICERM CCE Feedback

### ISSUE THREAD

**Issue by nilsdeppe** [sxs-collaboration/spectre#6246]
Here are notes I took at ICERM from feedback I got from various folks on how we could improve CCE usability.

- [x] A grid point-based approach would be helpful where we give people the grid point locations to dump and they can dump the metric and derivatives there. (#6398 )
   * Zach would like radial derivatives
   * ETK would like Cartesian derivatives.
- [x] We should have a glob-approach and then combine the HDF5 file on the fly so that people don’t need to do `combine-h5` or something to get a single file. One option would be to make the `combine-h5` distributed, but it would be even easier for users if we just do the combine before starting the CCE computation.
   1. Add HDF5 combining to `CceReduceWorldtube` by giving folks the ability to specify a glob for the file names. E.g. `Lev1_*/Run/CceOutputR0215.h5` (#6398 )
   2. There may be overlapping times in the files. We need to remove these. We should have an option to select times from either the earlier or later file. Ideally the data should be the same, but I don't think we need to check that. ETK uses first file. (partially #6298. This uses later file.)
- [ ] We need to document reasonable temporal sampling for the worldtube input. That is, how often should people dump worldtube data? We should provide some heuristic advice and comment on inspiral vs. merger since the time scales change. I would guess in SpEC and SpECTRE we get around this by dump some every `N` steps, but I don't know. @keefemitman @markscheel could you comment on how we decide how frequently to dump the worldtube?
- [ ] Could we compress the files? E.g. RPDMB, spectral fit in time? This is lower priority compared to the other points.
- [x] Document the Ylm convention we assume since there are different ones. This probably is just pointing to spherepack docs, roughly speaking. @nilsdeppe can add this. (#6271 )
- [x] We should provide an executable that dumps a text file of the theta & phi (or maybe 2 files) coordinates to use for a given ell. @knelli2 is working on this. (#6284 )
- [x] We should provide a C++ routine that takes in a given ell and returns the theta and phi points that need to be dumped to. @nilsdeppe will do this. The idea is to have a function in a test file that computes the points and then we have a snippet in the documentation. We can then test that the function returns the correct values for different ell. (#6271 )
- [ ] Rename `CharacteristicExtract` to `CharacteristicEvolution`
- [x] Rename `ReduceCceWorldtube` to `PreprocessCceWorldtube` (or something like that) (#6398 )
- [x] Deprecate reading anything other than modal Bondi data in `CharacteristicExtract` so that `ReduceCceWorldtube` needs to be run to get data for CCE. (#6299 )
- [ ] Check if we can chunk data on output of `ReduceCceWorldtube` for optimal performance of CCE
- [x] Support reading Cartesian derivatives of the lapse, shift, and spatial metric (nodal values) (#6397 )
- [ ] Support reading extrinsic curvature instead of time derivative of spatial metric, if this is significantly easier for other codes. Low priority and we should check with other codes before putting work in.

**Comment by knelli2** [comment]
3. SpEC observes [400 times per orbit](https://github.com/sxs-collaboration/spec/blob/80bf37f8c008d3b4e04b80c25b33e6897a1108f8/InputFiles/Bbh/DoMultipleRuns.input#L191).
